modules/core/src/main/java/org/apache/ignite/internal/client/impl/connection/GridClientNioTcpConnection.java

```diff
@@ -638,6 +644,29 @@ private void removePending(long reqId) {
             closedLatch.countDown();
     }
 
+    /** */
+    private <R> GridClientFutureAdapter<R> makeAuthRequest() throws GridClientConnectionResetException {
+        TcpClientFuture<R> fut = new TcpClientFuture<>();
+
+        fut.retryState(TcpClientFuture.STATE_REQUEST_RETRY);
+
+        GridClientAuthenticationRequest req = buildAuthRequest();
+
+        pendingReqs.putIfAbsent(req.requestId(), fut);
+
+        GridNioFuture<?> sndFut = ses.send(req);
+
+        try {
+            sndFut.get();
+        }
+        catch (Exception e) {
+            throw new GridClientConnectionResetException("Failed to send message over connection " +
+                "(will try to reconnect): " + serverAddress(), e);
+        }
+
+        return fut;
```

Comment:
Duplication of `makeRequest`, please reuse `makeRequest` instead.

Reply:
fixed